tests/portal-chooser/selftest-portal.sh

```
@@ -0,0 +1,142 @@
+      probe_rc=$?
+    fi
+
+    kill -INT "$portal_pid" 2>/dev/null
+    wait "$portal_pid" 2>/dev/null
+    portal_rc=$?
+
+    echo "PROBE_RC=$probe_rc"
+    echo "PORTAL_RC=$portal_rc"
+    echo "--- recorded calls ---"
+    cat "$log" 2>/dev/null || true
+  ' >"$work/case.out" 2>&1
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)